docs/changelog/4.2.rst

```diff
@@ -0,0 +1,1147 @@
+Changelogs for 4.2.x
+====================
+
+.. changelog::
+  :version: 4.2.0
+  :released: *unreleased*
+
+  This release *blah blah blah* …
+
+  Major features for this release:
+
+  * Lua records
+  * IXFRDist
+  * API changes
+  * DNSSEC? (#6655)
+
+  Minor features, weave paragraph from this:
+
+  * 6823: On incoming NOTIFY load our serial from backend to have it
+  * 6969: re-do of an earlier PR to add the glorious log-log histograms
+  * 6997: Name threads in the programs
+  * 6727: ALIAS: Respond SERVFAIL on non-NOERRORs from resolver
+  * 6910: support for OpenSSL 1.1.1's ed25519 and ed448
+  * 6975: Enhance query-logging with timing
+  * 6913: actually truncate truncated responses
+  * 6730: Implement a smarter dedup for filling packets in auth
+
+  Upgrade notes:
+
+  * autoserial removed
+  * command line util parameters (for scripts)
+  * BIND config files fixup on package upgrade
+  * GOST and Botan removed
+  * #6633?
+
+  Possible tags we can use below: Bug Fixes, Improvements, Internals, API, Packages, PostgreSQL, LDAP, MySQL, Tools, DNSSEC, Removed Features, New Features, BIND, DNSUpdate, IXFRDist.
+
+  .. change::
+    :tags: Improvements, Internals
+    :pullreq: 7245
+
+    Use a cache-able soa record for the serial check caused by a notify.
+
+  .. change::
```

Comment:
What do you think of grouping major feature related changes ? "Improving" here lua records seems a bit off context as it is a new feature part of this release.

Reply:
We could do this for all other major features such as ixfrdist.

Reply:
Above all the separate changelog entries (where the bullet list reminders / suggestions are now) we will have a description talking about the major features of the 4.2.x branch.

We could group the individual changelog entries, it would be nice, but it's a lot of work.